include/ensmallen_bits/cmaes/pop_cmaes_impl.hpp

```diff
@@ -0,0 +1,168 @@
+
+  overallSBC = sbc;
+  ElemType objective;
+  size_t evaluations;
+
+  size_t defaultLambda = this->PopulationSize();
+  size_t currentLargeLambda = defaultLambda;
+
+  double stepSizeDefault = this->StepSize();
+
+  // Print out the default population size
+  Info << "Default population size: " << defaultLambda << std::endl;
+
+  size_t restart = 0;
+
+  while (restart < maxRestarts)
+  {
+    if (!UseBIPOPFlag || largePopulationBudget <= smallPopulationBudget || 
+        restart == 0 || restart == maxRestarts - 1)
+    {
+      // Large population regime (IPOP or BIPOP)
+      currentLargeLambda *= populationFactor;
+      this->PopulationSize() = currentLargeLambda;
+      this->StepSize() = stepSizeDefault;
+
+      Info << "POP-CMA-ES: restart " << restart << ", large population size" <<
+          " (lambda): " << this->PopulationSize() << std::endl;
+      
+      iterate = iterateIn;
+
+      // Optimize using the CMAES object.
+      objective = CMAES<SelectionPolicyType, 
+          TransformationPolicyType>::Optimize(function, iterate, sbc, 
+          callbacks...);
+
+      evaluations = this->FunctionEvaluations();
+      largePopulationBudget += evaluations;
+    }
+    else if (UseBIPOPFlag)
+    {
+      // Small population regime (BIPOP only)
+      double u = dis(gen);
+      size_t smallLambda = static_cast<size_t>(defaultLambda * std::pow(0.5 * 
+          currentLargeLambda / defaultLambda, u * u));
+      double stepSizeSmall = 2 * std::pow(10, -2*dis(gen));
+      
+      this->PopulationSize() = smallLambda;
+      this->StepSize() = stepSizeSmall;
+
+      Info << "BIPOP-CMA-ES: restart " << restart << ", small population" <<
+          " size (lambda): " << this->PopulationSize() << std::endl;
+
+      iterate = iterateIn;
+      
+      // Optimize using the CMAES object.
+      objective = CMAES<SelectionPolicyType, 
+          TransformationPolicyType>::Optimize(function, iterate, sbc, 
+                                              callbacks...);
+
+      evaluations = this->FunctionEvaluations();
+      smallPopulationBudget += evaluations;
+    }
+
+    if (objective < overallObjective)
+    {
+      overallObjective = objective;
+      overallSBC = sbc;
+      Info << "POP-CMA-ES: New best objective: " << overallObjective << 
+          std::endl;
+    }
+
+    totalFunctionEvaluations += evaluations;
+    // Check if the total number of evaluations has exceeded the limit
+    if (totalFunctionEvaluations >= maxFunctionEvaluations) {
+      Warn << "POP-CMA-ES: Maximum function overall evaluations reached. "
+           << "terminating optimization." << std::endl;
+
+      Callback::EndOptimization(*this, function, iterate, callbacks...);
+      iterateIn = overallSBC.BestCoordinates();
+      return overallSBC.BestObjective();
+    }
+
+    ++restart;
+  }
+
+  Callback::EndOptimization(*this, function, iterate, callbacks...);
+  iterateIn = overallSBC.BestCoordinates();
```

Comment:
```suggestion
  iterateIn = std::move(overallSBC.BestCoordinates());
```
Another copy we can avoid.